### ISSUE TITLE
[Feature] Merged allow/deny configuration like `.cargo/config.toml`

### ISSUE THREAD

**Issue by bes** [rust-lang/rust-clippy#14922]
### Description

I have a feature request.

For "monorepos" i.e. a git with multiple workspace / non-workspace crates, it would be very convenient to configure all lints at a top level, much like `.cargo/config.toml` configures cargo.

Since https://github.com/rust-lang/rust-clippy/issues/6625 / https://github.com/rust-lang/cargo/issues/12115 it became possible to configure workspace-level lints and then opt-in to the linting in each workspace member, but even that is a bit verbose and tedious for large monorepos. Consider how https://github.com/emilk/egui/pull/4413/files has 72 instances of `[lints] workspace = true`.

In my particular case I have a monorepo with 4 workspaces and 5 applications, meaning I have to repeat the contents of `[workspace.lints.clippy]` and `[lints.clippy]` 9 times, in addition to the 40+ times I have to repeat `[lints] workspace = true`.

The repetition isn't the issue, the issue is rather the need for all contributors to remember to update all 9 `[...lints.clippy]` declarations when those settings need to change.

Furthermore it would be amazing if such a configuration file could be able to affect the configuration flags of clippy, i.e. `--all-features --all-targets -- -D warnings` _once_ so that it wouldn't have to be repeated endlessly in `.gitlab-ci.toml`, `bacon.toml`, and various other shell scripts.

If all of this is already possible and handy, I apologize.

TL;DR:
* Mergeable configuration for Clippy much like `.cargo/config.toml`
* Possibility to configure allow/deny/lint groups
* Possibility to configure "command line options" such as `--all-features --all-targets -- -D warnings`

Thank you for an amazing tool!

### Version

```text

```

**Comment by Centri3** [comment]
For now (ab?)using rustflags could likely work for you (despite the fact that feature is supposed supersede that)

I'm honestly a bit surprised clippy's config doesn't already let you do this. Was there discussion on this in the past?

**Comment by bes** [comment]
Thanks, that worked, but has a few issues for my setup:

* Tweaking anything in `RUSTFLAGS` means rebuilding the world which isn't... optimal =)
* One of the non-workspace crates is a wasm library which sets specific rustflags for `[target.wasm32-unknown-unknown]`, so the lints aren't applied there unless i duplicate them there, so kind of back to square one on that one!

**Comment by Centri3** [comment]
Yeah those annoyances are practically the reasons this was added :P No real way around it afaik

**Comment by bes** [comment]
The irony is I prefer it to the workspace lints since it's less work.

**Comment by bes** [comment]
@Centri3 Before / if I give it an attempt, could you outline what the challenges are? I assume there will be some pusback on this since there are already other ways (Workspace, RUSTFLAGS, command line params) to achieve the same thing.

Note to self: I just realized using RUSTFLAGS for all of this has the unfortunate side effect of not being able to do builds with warnings (while developing) since those will just instantly become build errors. I have to remove `"-D", "warnings",` from RUSTFLAGS and then add that back for each invocation instead.

**Comment by bes** [comment]
Of all the suboptimal ways of achieving this (rustflags, RUSTFLAGS, the `cfg(all())` hack, etc. etc.) maybe this is the best I've found yet: Just don't use any of those features and wrap everything in this dumb little script:

```shell
#!/bin/bash

cargo clippy "$@" -- \
    -D warnings -D clippy::all \
    -A clippy::too_many_arguments \
    -A clippy::new_without_default \
    -A clippy::needless_lifetimes \
    -A clippy::question-mark \
    -A clippy::let-and-return
```

Runs clippy with whatever flags you give the script and then adds your wanted allow flags on top
